### ISSUE TITLE
Spooky object insertion in cross-after-aggregate with autopadding

### ISSUE THREAD

**Issue by Jerrythafast** [vega/vega#439]
Dear Vega devs,

Here I am again (sorry, guys...) with probably the spookiest issue I have filed so far. The MWE:
```json
{
  "data": [
    {
      "name": "a",
      "values": [1, 2, 3]
    },
    {
      "name": "b",
      "values": [0, 1]
    },
    {
      "name": "c",
      "source": "a",
      "transform": [
        {
          "type": "aggregate",
          "summarize": {"data": "max"}
        },
        {
          "type": "formula",
          "field": "surprise",
          "expr": "'surprise!'"
        },
        {
          "type": "cross",
          "with": "b",
          "filter": "datum.b.data == 0"
        }
      ]
    },
    {
      "name": "d",
      "source": "a",
      "transform": [
        {
          "type": "aggregate",
          "summarize": {"data": "max"}
        },
        {
          "type": "cross",
          "with": "b",
          "filter": "datum.b.data == 1"
        }
      ]
    }
  ]
}
```

Can you guess what you get if you do `ved.view.data("d").values()` with the above spec? Here comes...

```json
[
  {
    "_id": 12,
    "a":   {"_id": 11, "max_data": 3},
    "b":   {"_id":  5, "data": 1}
  },
  {
    "_id": 14,
    "a":   {"_id":  9, "max_data": 3, "surprise": "surprise!"},
    "b":   {"_id":  5, "data": 1}
  }
]
```

Interestingly, `ved.view.data("c").values()` is fine...

```json
[
  {
    "_id": 10,
    "a":   {"_id":  9, "max_data": 3, "surprise": "surprise!"},
    "b":   {"_id":  4, "data": 0}
  }
]
```

But even more interestingly, ***this only happens when using auto-padding***. If you add padding amounts to the spec, it works as expected unless you break it again by doing `ved.view.padding("auto").update()`.

Kind regards,
Jerry

_(Note: The given MWE triggers #395 in v2.3.1, so you'll need to get an up-to-date build of Vega to test this.)_

**Comment by arvind** [comment]
If we weren't [already inclined](https://github.com/vega/vega/issues/424#issuecomment-155122183) to award your intrepid bug finding, this issue certainly takes the cake! This issue spurred bug fixes across _three_ Vega projects:

* datalib's aggregator uses a series of flags to identify which aggregate cells should be included in the changeset. But, this would result in reflow tuples being swallowed up if they did not trigger a reevaluation. Fixed with vega/datalib@1452084 (v1.4.10) and 91552c8.
* The pollution you were seeing was because vega-dataflow's Collectors were augmenting the input changeset they receive. This is a problem when the Graph merges reflow and regular changesets during propagation. vega/vega-dataflow@2488858 (v1.3.1) corrects this.

Thanks for a great bug report, @Jerrythafast. I really appreciate that you're able to distill problematic specs down to MWE. Makes our lives _so_ much easier.

**Comment by Jerrythafast** [comment]
Thanks for the swift fix, guys! (Shouldn't it be past midnight where you live?)

In https://github.com/vega/vega/commit/91552c8e078d7c99ed299c6ccdbb33344fb8f591 you have increased the required datalib version number to v1.4.10, but dataflow is still at 1.3.0. Is there a reason for this?
